framework/src/fvkernels/FVElementalKernel.C

```diff
@@ -59,17 +59,43 @@ FVElementalKernel::computeResidual()
 void
 FVElementalKernel::computeJacobian()
 {
+  const auto r = computeQpResidual() * _assembly.elemVolume();
+
+#ifdef MOOSE_GLOBAL_AD_INDEXING
+  // When doing global indexing, we process *all* the derivatives/Jacobian entries here, not just
+  // the on-diagonal ones. By doing this we avoid doing dof index lookups to determine what
+  // derivative entries actually correspond to the on-diagonal.
+  const auto & derivs = r.derivatives();
+
+  const auto & column_indices = derivs.nude_indices();
+  const auto & values = derivs.nude_data();
+
+  mooseAssert(column_indices.size() == values.size(), "Indices and values size must be the same");
+
+  const auto & row_indices = _var.dofIndices();
+  mooseAssert(row_indices.size() == 1, "We're currently built to use CONSTANT MONOMIALS");
```

Comment:
Who is "we" in all of these asserts?

Reply:
Yeah - this should read more like "How TF do *you* have more than one dof index with FV only supporting const monomials?".

Reply:
> Who is "we" in all of these asserts?

you